### PR TITLE
Check for K8S GW CRDs before applying waypoint

### PR DESCRIPTION
**Pull request by ymesika**
**Please provide a description of this PR:**

Mainly to have a better error message for the users if the CRDs don't exist.

### CODE REVIEW

File: istioctl/cmd/waypoint.go [hunk 1]
```diff
@@ -102,6 +102,12 @@ func waypointCmd() *cobra.Command {
 			if err != nil {
 				return fmt.Errorf("failed to create Kubernetes client: %v", err)
 			}
+			if _, err = client.Ext().ApiextensionsV1().CustomResourceDefinitions().Get(context.Background(),
+				"gateways.gateway.networking.k8s.io", metav1.GetOptions{}); err != nil {
```
Comment (hanxiaop) [inline istioctl/cmd/waypoint.go:106]: Can you use this variable? https://github.com/istio/istio/blob/master/pkg/config/schema/gvk/resources.gen.go#L22

**Review by howardjohn — COMMENTED** [review]
Can we check the error message after apply if it fails?

It will be faster and more correct (reading CRDs like this is not technically "correct" in k8s)

**Review by hanxiaop — APPROVED** [review]

File: istioctl/cmd/waypoint.go [hunk 2]
```diff
@@ -112,6 +114,9 @@ func waypointCmd() *cobra.Command {
 				FieldManager: "istioctl",
 			})
 			if err != nil {
+				if strings.Contains(err.Error(), missingResourceMessageFragment) {
```
Comment (howardjohn) [inline istioctl/cmd/waypoint.go:117]: nit: you can use `errors.IsNotFound`. I know we do this same check elsewhere though

**Review by howardjohn — COMMENTED** [review]
LGTM

File: istioctl/cmd/waypoint.go (hunk 2 above)
Comment (ymesika) [reply istioctl/cmd/waypoint.go:117]: That's much cleaner. I think it's not a nit.

**Pull request merged**